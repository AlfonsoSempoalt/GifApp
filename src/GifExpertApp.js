import React, { useState } from "react";
import CategoryAdd from "./components/CategoryAdd";
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories=[] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h1>GifApp</h1>
      <CategoryAdd setCategories={setCategories} />
      <hr/>
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
