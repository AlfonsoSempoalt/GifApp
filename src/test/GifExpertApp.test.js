import { describe, expect } from '@jest/globals';
import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('tests to <GifExpertApp />', () => {
  const categories = ['One punch man', 'One piece'];
  let wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

  beforeEach(() => {
    wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
  });

  test('should match with the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should renderize two categories(<GifGrid />)', () => {
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
