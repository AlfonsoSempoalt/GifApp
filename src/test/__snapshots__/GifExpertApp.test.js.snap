// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests to <GifExpertApp /> should match with the snapshot 1`] = `
<Fragment>
  <h1>
    GifApp
  </h1>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch man"
      key="One punch man"
    />
    <GifGrid
      category="One piece"
      key="One piece"
    />
  </ol>
</Fragment>
`;
